### PR TITLE
Update header of sqlite3 to 3.28.0

### PR DESCRIPTION
**Pull request by ErnyTech**
Since the SQLite 3 bindings have been outdated since about 2016, I decided to update them to the latest version of SQLite available
This is a work in progress and I would be grateful if someone with more experience than me about CAPI3REF could analyze this update attempt, thanks

### CODE REVIEW

**Comment by CyberShadow** [comment]
Thanks a lot! It looks good.

Can you describe the process you used to perform these changes, so that it can be useful for future updates?

**Comment by ErnyTech** [comment]
I simply analyzed by hand the new header with that of the old version through the "diff" tool, then I applied the changes to the D header and read all two headers to analyze discrepancies.

It is tedious to do but not difficult, after all they are only about 2k of simple code

**Comment by ErnyTech** [comment]
I did not use any automatic tool for generation of D headers

**Comment by CyberShadow** [comment]
Thanks. Is there anything else you wanted to do, considering the "WIP" status?

**Comment by ErnyTech** [comment]
I'm going to check the code again as soon as I have some time

**Comment by andre2007** [comment]
Also quite relevant due to vulnerability 
https://www.securityweek.com/remote-code-execution-vulnerability-impacts-sqlite

Thanks for your work

**Comment by CyberShadow** [comment]
As I mentioned several times in #6867 , SQLite has a stable ABI, and the bindings are independent to the implementation. This pull request is welcome, but completely orthogonal.

**Comment by ErnyTech** [comment]
@andre2007 This pull request does not solve in any way any problem of vulnerability, it is not an update to the implementation of sqlite3, it is a simple update to the header to interface to the abi of sqlite3.

The usefulness of this pull request is only the possibility of using the new abi of sqlite3, no change in terms of security

**Comment by ErnyTech** [comment]
Now the patch can be reviewed, it is no longer in wip

File: etc/c/sqlite3.d [hunk 1]
```diff
@@ -317,39 +333,60 @@ struct sqlite3_io_methods
...
+    SQLITE_FCNTL_PDB                     = 30,
+    SQLITE_FCNTL_BEGIN_ATOMIC_WRITE      = 31,
+    SQLITE_FCNTL_COMMIT_ATOMIC_WRITE     = 32,
+    SQLITE_FCNTL_ROLLBACK_ATOMIC_WRITE   = 33,
+    SQLITE_FCNTL_LOCK_TIMEOUT            = 34,
+    SQLITE_FCNTL_DATA_VERSION            = 35,
+    SQLITE_FCNTL_SIZE_LIMIT              = 36
+}
+
+deprecated ("deprecated names")
```
[the remark this answers is not in the document]
Comment (ErnyTech) [reply etc/c/sqlite3.d]: The deprecated attribute has already been used previously in this header:
https://github.com/dlang/phobos/blob/9e8086afb38ff2faa7bec4021cb03ccf1735a75f/etc/c/sqlite3.d#L1003

https://github.com/dlang/phobos/blob/9e8086afb38ff2faa7bec4021cb03ccf1735a75f/etc/c/sqlite3.d#L1296

File: etc/c/sqlite3.d [hunk 2]
```diff
@@ -472,36 +509,50 @@ enum
...
     SQLITE_CONFIG_PAGECACHE            = 7,  /** void*, int sz, int N */
     SQLITE_CONFIG_HEAP                 = 8,  /** void*, int nByte, int min */
     SQLITE_CONFIG_MEMSTATUS            = 9,  /** boolean */
     SQLITE_CONFIG_MUTEX                = 10,  /** sqlite3_mutex_methods* */
     SQLITE_CONFIG_GETMUTEX             = 11,  /** sqlite3_mutex_methods* */
 /* previously SQLITE_CONFIG_CHUNKALLOC 12 which is now unused. */
-    SQLITE_CONFIG_LOOKASIDE            = 13,  /** int int */
-    SQLITE_CONFIG_PCACHE               = 14,  /** sqlite3_pcache_methods* */
-    SQLITE_CONFIG_GETPCACHE            = 15,  /** sqlite3_pcache_methods* */
+    SSQLITE_CONFIG_LOOKASIDE           = 13,  /** int int */
```
[the remark this answers is not in the document]
Comment (ErnyTech) [reply etc/c/sqlite3.d]: Fixed

**Comment by ErnyTech** [comment]
how do i restart the tests?  have solved the problem with https://github.com/dlang/phobos/pull/7011

**Comment by thewilsonator** [comment]
Force push with no changes.

**Pull request merged**